src/chapter03-getting-started.md

```diff
@@ -110,27 +110,75 @@ Note that the echo request in step 6 is just a simple usage example. Once the co
     2. Allocate and initialize a comm server configuration structure using the transport configuration from step 1.1
-2. Initialize the server NVM context
-    1. Allocate and initialize a config, context, and callback structure for the NVM flash storage drivers (the implementation of these structures is provided by the port)
-    2. Allocate and initialize an NVM config structure using the NVM flash configuration from step 2.1
-    3. Allocate an NVM context structure and initialize it with the configuration from step 2.2 using `wh_Nvm_Init()`
-3. Allocate and initialize a crypto context structure for the server
-4. Initialize wolfCrypt (before initializing the server)
-5. Allocate and initialize a server config structure and bind the comm server configuration, NVM context, and crypto context to it
-6. Allocate a server context structure and initialize it with the server configuration using `wh_Server_Init()`
-7. Set the server connection state to connected using `wh_Server_SetConnected()` when the underlying transport is ready to be used for client communication (see [TODO](todo) for more information)
-8. Process client requests using `wh_Server_HandleRequestMessage()`
+2.  Allocate and initialize a crypto context structure for the server
+3. Initialize wolfCrypt (before initializing the server)
+4. Allocate and initialize a server config structure and bind the comm server configuration, NVM context, and crypto context to it
+5. Allocate a server context structure and initialize it with the server configuration using `wh_Server_Init()`
+6. Set the server connection state to connected using `wh_Server_SetConnected()` when the underlying transport is ready to be used for client communication (see [TODO](todo) for more information)
+7. Process client requests using `wh_Server_HandleRequestMessage()`
+
+The server may be configured to support NVM object storage using NVM flash configuration. Include the steps to [initialize NVM](./chapter04-functional-components.md#NVM-Architecture) on the server after step 1.
 
 ```c
  /* TODO: basic server configuration example */
+#include <string.h> /* for memcmp() */
+#include "wolfhsm/server.h"  /* Server API (includes comm config) */
+#include "wolfhsm/wh_transport_mem.h" /* transport implementation */
+
+/* Step 1.1: Allocate and initialize the shared memory transport configuration */
+/* Shared memory transport configuration */
+static whTransportMemConfig transportMemCfg = { /* shared memory config */ };
+
+/* Shared memory transport context (state) */
+whTransportMemServerContext transportMemServerCtx = {0};
+
+/* Callback structure that binds the abstract comm transport interface to
+ * our concrete implementation */
+whTransportServerCb transportMemServerCb = {WH_TRANSPORT_MEM_SERVER_CB};
+
+/* Step 1.2: Allocate a comm server configuration structure and bind to the
+ * transport */
+/* Configure the server comms to use the selected transport configuration*/
+whCommServerConfig commServerCfg[1] = {{
```

Comment:
could you remove the "array of 1" syntax for these examples (e.g. `whStruct myStruct` instead of `whStruct myStruct[1]`) pls? And check the client example above too(I think I forgot to remove one). It is nice because you don't need `&` but some people may be unfamiliar with that syntax and we should just do it the traditional way for examples. You will need to ensure all the pointer args to the functions now take their struct parameters with an `&` to get the address.